workbench-core/authorization/src/dynamicAuthorization/dynamicAuthorizationService.test.ts

```diff
@@ -131,26 +144,146 @@ describe('DynamicAuthorizationService', () => {
     jest.resetAllMocks();
   });
 
-  describe('init', () => {
-    it('throws a not implemented exception', async () => {
-      await expect(dynamicAuthzService.init({})).rejects.toThrow(Error);
+  describe('isAuthorizedOnSubject', () => {
+    let mockDynamicOperation: DynamicOperation;
+    let params: IsAuthorizedOnSubjectRequest;
+    beforeEach(() => {
+      mockUser = {
+        id: 'sampleId',
+        roles: ['sampleGroupId']
+      };
+      mockDynamicOperation = {
+        action: sampleAction,
+        subject: {
+          subjectType: sampleSubjectType,
+          subjectId: sampleSubjectId
+        }
+      };
+      auditAction = 'isAuthorizedOnSubject';
+      params = {
+        authenticatedUser: mockUser,
+        dynamicOperation: mockDynamicOperation
+      };
+    });
+    test('is authorized on subject', async () => {
+      mockDynamicAuthorizationPermissionsPlugin.getIdentityPermissionsBySubject = jest
+        .fn()
+        .mockResolvedValueOnce({
+          data: {
+            identityPermissions: mockIdentityPermissions
+          }
+        })
+        .mockResolvedValueOnce({
+          data: {
+            identityPermissions: []
+          }
+        });
+      await dynamicAuthzService.isAuthorizedOnSubject(params);
```

Comment:
should we be testing that this doesnt throw?

Reply:
and/or the output?

Reply:
`isAuthorizedOnSubject` is returning `Promise<void>` and since there is `await` the test will fail if unexpected error is thrown so explicit `not.toThrow()` is IMO redundant.

Reply:
Its more to show that this function is under test. Putting an expect there makes it obvious we are testing this function and we are explicitly stating that it should not throw.